### ISSUE TITLE
Pi issue

### ISSUE THREAD

**Issue by andrew222651** [Pomax/tex2utf#9]
Input:

`$\pi + \Pi$`

Output:

`\pi + π`

We'd want `π + Π`, right?

**Comment by Pomax** [comment]
pretty sure we do, but unfortunately I'm also going to have to point to this: https://github.com/Pomax/tex2utf#how-does-this-script-work

(there is nothing easy or simple about the original .pl source, and while I'm more than happy to accept PRs, unless it's affecting formulae I personally need typeset, I can't justify the code archaeology required to "fix" things, given that I have no idea how the original script author figured things should work =)

**Comment by oylenshpeegul** [comment]
It looks like \Pi is just set to be π. I changed it to \pi and added a \Pi that is Π

https://github.com/oylenshpeegul/tex2utf/commit/b37244301c08c518b0b063d5df011d31a177bf68
    
But that ate the space before the plus sign

```
❯ ./tex2utf.pl <<< '$\pi + \Pi$'
π+ Π
```
Jamming in almost any TeX space there seems to fix it

```
❯ ./tex2utf.pl <<< '$\pi~+ \Pi$'
π + Π
```
but is that normal?

**Comment by Pomax** [comment]
In terms of TeX: π is a normal symbol that is treated the same as any other variable in a maths formula: no spaces around it, it joins up with whatever regular terms/symbols are next to it. The product symbol, however, _must_ have spaces around it because that's how we write products. So I would say that `π+ Π` is entirely expected.

However, that's also just a guess: as per the previous comment this is a port of _very old perl code_ and figuring out what it "should" do is a non-trivial task. If you know someone who still speaks fluent perl 4 as well as modern python 3, send them my way, hopefully we can move this over to something modern and easy to maintain instead =)

**Comment by oylenshpeegul** [comment]
But isn't \Pi just a capital pi in TeX? We say \prod for the product symbol. So \pi and \Pi ought to be treated the same.

It seems to have something to do with being first.

```
❯ ./tex2utf.pl <<< '\Pi + \Pi'
Π+ Π

❯ ./tex2utf.pl <<< '\prod + \prod'
___  ___
│ │+ │ │
```

**Comment by Pomax** [comment]
Good point. 

I _literally_ have no idea how this script works though, I just updated it for my own needs enough to make it look good using modern unicode, so if you have a PR that fixes it: oh my god please file it against this repo and I will merge it in without question.

**Comment by sujaldev** [comment]
@Pomax I'm confused, you're not the author of this script?

**Comment by Pomax** [comment]
Correct, all I did was take the over 20 years old `tex2mail` script, and modify it just enough so that it'll run in Unicode mode using Perl 5 (rather than ASCII in the now long dead Perl 4), with some obvious replacements for individual characters that have modern correct Unicode equivalents, and some "hopefully this works" replacements for multi-row symbols.
